_plugins/bidirectional_links_generator.rb

```diff
@@ -7,14 +7,14 @@ def find_dups
 # frozen_string_literal: true
 class BidirectionalLinksGenerator < Jekyll::Generator
   def generate(site)
-    graph_nodes = []
-    graph_edges = []
-    crossfit_graph_nodes = []
-    crossfit_graph_edges = []
-    design_graph_nodes = []
-    design_graph_edges = []
-    nutrition_graph_nodes = []
-    nutrition_graph_edges = []
+    sub_graph_nodes = {}
+    sub_graph_edges = {}
+
+    tags = site.config["graph"]
+    tags.each do |tag|
+      sub_graph_nodes.merge!({"#{tag}": []})
+      sub_graph_edges.merge!({"#{tag}": []})
+    end
```

Comment:
grabbing the site config and creating a hash for each tag it finds in nodes and edges